src/test/scala/sigmastate/CostingSpecification.scala

```diff
@@ -197,9 +197,21 @@ class CostingSpecification extends SigmaTestingData {
     val selfTree = "SELF.R6[AvlTree].get"
     val sizeOfArgs = Seq(sizeOf(avlTree), sizeOf(key1), sizeOf(lookupProof)).foldLeft(0L)(_ + _.dataSize)
     val containsCost = perKbCostOf(sizeOfArgs, avlTreeOp)
-    
+
     cost(s"{ $selfTree.contains(key1, lookupProof) }") shouldBe (AccessTree + containsCost + constCost)
     cost(s"{ $selfTree.get(key1, lookupProof).isDefined }") shouldBe (AccessTree + containsCost + constCost + selectField)
     cost(s"{ $selfTree.getMany(keys, lookupProof).size > 0 }") shouldBe (AccessTree + containsCost + constCost + LengthGTConstCost)
   }
+
+  property("Option operations cost") {
+    val opt = "SELF.R5[Int]"
+    val accessOpt = accessBox + accessRegister
+    cost(s"{ $opt.get > 0 }") shouldBe (accessOpt + selectField + GTConstCost)
+    cost(s"{ $opt.isDefined }") shouldBe (accessOpt + selectField)
+    cost(s"{ $opt.getOrElse(1) > 0 }") shouldBe (accessOpt + selectField + GTConstCost)
+    // TODO (45 != 48)  cost(s"{ $opt.filter({ (x: Int) => x > 0}).isDefined }") shouldBe
+    //      (accessOpt + OptionOp + lambdaCost + GTConstCost + selectField)
+    // TODO (45 != 48)  cost(s"{ $opt.filter({ (x: Int) => x > 0}).isDefined }") shouldBe
+    //      (accessOpt + OptionOp + lambdaCost + GTConstCost + selectField)
+  }
```

Comment:
@greenhat what about this tests?

Reply:
Fixed in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/469/commits/fee52bab7c1205a3f51add12b7a65953dc8fe1b6